libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncUpTarget.java

```diff
@@ -245,24 +255,64 @@ public int updateOnServer(SyncManager syncManager, String objectType, String obj
      * @param record
      * @return
      */
-    public String fetchLastModifiedDate(SyncManager syncManager, JSONObject record) {
-        try {
-            final String objectType = (String) SmartStore.project(record, Constants.SOBJECT_TYPE);
-            final String objectId = record.getString(getIdFieldName());
+    protected RecordModDate fetchLastModifiedDate(SyncManager syncManager, JSONObject record) throws JSONException, IOException {
```

Comment:
Called from isNewerThanServer which used to live in SyncManager but now lives in SyncUpTarget (made protected for that reason).
Also the return value is richer (it's the time stamp + whether the record was deleted), which allows us to be a bit more precise (we won't even try an update for a remotely deleted record if leave-if-changed).